### DISCUSSION TITLE
Issue with Winget Autoupdate

### DISCUSSION THREAD

**Discussion by DeGrootePasja** [Romanitho/Winget-AutoUpdate#662] [General]
Hi everyone,

I recently installed Winget AutoUpdate and configured a whitelist for selective updates. However, when I attempt to update the applications listed in the whitelist using the command ./Winget-AutoUpdate-Install.ps1 -UseWhiteList, it fails to update the specified tool (7zip.7zip). Or is it that there isn't a manual trigger possible?

Has anyone encountered this issue before, or can provide any insight into what might be going wrong? Any assistance would be greatly appreciated!

Thank you

**Comment by Romanitho** [comment]
Hi.
Not sure to understand what you try to do. If you have installed WAU with a whitelist, and the app is in your list, it should update your app.
Can you share the log file please?

**Reply by DeGrootePasja** [reply]
Hi, 

The app's ID is in the included_apps.txt file (7zip.7zip) & these are the log files ( I think those are the ones you meant )

"WAU-Installer.log"
`################################################################
 21/08/2024 -   INSTALLING WAU
################################################################
12:00:50 - Checking prerequisites...
12:00:50 - Prerequisites checked. OK

12:00:50 - Checking if WinGet is installed/up to date.
12:00:51 - -> WinGet is up to date: v1.8.1911

12:00:51 - Installing WAU 1.20.1...
12:00:51 - -> Updating previous installation...
12:00:51 - -> Copied a brand new Whitelist.
12:00:53 - -> Installing WAU scheduled tasks
12:00:53 - -> Setting Registry config
12:00:53 - -> Checking Mods Directory:
12:00:53 -    The mods directory is secured!
12:00:53 - -> Checking Functions Directory:
12:00:53 -    The Functions directory is secured!
12:00:54 - -> WAU Installation succeeded!

12:01:02 - Running Winget-AutoUpdate...
12:01:10 - End of process.`

"updates.log"
`################################################################
 21/08/2024 - CHECK FOR APP UPDATES (System context - Connected user)
 ################################################################
 12:01:06 - Notification Level: Full. Notification Language: English
 12:01:06 - Checking internet connection...
 12:01:06 - Connected !
 12:01:06 - Metered connection detected.
 12:01:06 - WAU is configured to bypass update checking on metered connection
 12:01:06 - End of process`

Additionally, my whitelist includes the ID "7zip.7zip". Since I'm not very experienced with WAU, I'm wondering if there might be something I've overlooked or misconfigured.

Thanks in advance.

**Reply by Romanitho** [reply]
```
12:01:06 - Metered connection detected.
12:01:06 - WAU is configured to bypass update checking on metered connection
```
Are you in a sharing mode? Please, use Ethernet, WiFi or allow WAU to run on metered connection :)

**Reply by DeGrootePasja** [reply]
The metered connection is turned of. But does it has to something with it being a Virtual machine? 

Sharing is turned off, just saw it.

Thanks in advance!

**Reply by Romanitho** [reply]
Not related to VM, I use VM to test WAU and new features.
Sharing mode, is when you share your smartphone 4G/5G data with your computer for example. To avoid any cost, we do not run WAU on metered connexion by default. But you can force WAU to use Data anyway.

**Reply by DeGrootePasja** [reply]
We have a wired connection to our VM, without any limitation & we can see that Metered connection is turned of in the settings.

**Reply by Romanitho** [reply]
just allow metered connection, it might be a false detection. If you use the GUI to install WAU, tick the checkbox. If you use the cli, use -RunOnMetered

**Reply by DeGrootePasja** [reply]
Thank you so much for all the help, it worked! I'm looking forward to using the tool

**Reply by Romanitho** [reply]
no worries ;)

**Reply by DeGrootePasja** [reply]
I've got another question. We now have ran it with a blacklist. The stuff in the blacklist wasn't updated (great), but there is also stuff that isn't in the blacklist that wasn't updated.

**Reply by Romanitho** [reply]
Log file? 😅
I can't tell like this. It can be an issue with WAU, with winget, with the app itself....

**Reply by DeGrootePasja** [reply]
Hi Romanitho,

Thanks for helping us sort out the issue by directing us to the log file—that was a huge help!

However, we've encountered another hiccup. When trying to update, our log indicates that there's already another version installed, and we're not quite sure how to proceed. Here's what the update.log file shows:
``
09:45:02 - ##########   WINGET UPGRADE PROCESS STARTS FOR APPLICATION ID 'Microsoft.Teams.Classic'   ##########
09:45:02 - -> Running: Winget upgrade --id Microsoft.Teams.Classic -e --accept-package-agreements --accept-source-agreements -s winget -h
  ███████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 2.61 MB
  ██████████████████████▒▒▒▒▒▒▒▒  2.00 MB / 2.61 MB
  ██████████████████████████████  2.61 MB / 2.61 MB
Found Microsoft Teams classic [Microsoft.Teams.Classic] Version 1.7.00.21751
This application is licensed to you by its owner. Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Starting package install...
Another version of this application is already installed.
Installer failed with exit code: 1638
Installer log is available at: <log_file_location>.log
09:45:05 - -> An upgrade for Teams Machine-Wide Installer failed, now trying an install instead... (1/2)
09:45:05 - -> Running: Winget install --id Microsoft.Teams.Classic -e --accept-package-agreements --accept-source-agreements -s winget -h --force
  ███████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 2.61 MB
  ██████████████████████▒▒▒▒▒▒▒▒  2.00 MB / 2.61 MB
  ██████████████████████████████  2.61 MB / 2.61 MB
Found Microsoft Teams classic [Microsoft.Teams.Classic] Version 1.7.00.21751
This application is licensed to you by its owner. Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Starting package install...
Another version of this application is already installed.
Installer failed with exit code: 1638
Installer log is available at: <log_file_location>.log
09:45:08 - -> An upgrade for Teams Machine-Wide Installer failed, now trying an install instead... (2/2)
09:45:08 - -> Running: Winget install --id Microsoft.Teams.Classic -e --accept-package-agreements --accept-source-agreements -s winget -h --force
  ███████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 2.61 MB
  ██████████████████████▒▒▒▒▒▒▒▒  2.00 MB / 2.61 MB
  ██████████████████████████████  2.61 MB / 2.61 MB
Found Microsoft Teams classic [Microsoft.Teams.Classic] Version 1.7.00.21751
This application is licensed to you by its owner. Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Starting package install...
Another version of this application is already installed.
Installer failed with exit code: 1638
Installer log is available at: <log_file_location>.log
09:45:11 - ##########   WINGET UPGRADE PROCESS FINISHED FOR APPLICATION ID 'Microsoft.Teams.Classic'   ##########
09:45:11 - Teams Machine-Wide Installer update failed.
09:45:16 - XAMPP : Skipped upgrade because it is in the excluded app list
09:45:16 - No new update.
09:45:16 - End of process!
``

Any insights or advice on how we might resolve this would be incredibly valuable.
Thanks in advance!

**Comment by AndrewDemski-ad-gmail-com** [comment]
That 1638 error is still open in MS winget project:
https://github.com/microsoft/winget-cli/issues/1971#issue-1148664203
In the meantime, please try removing 'Microsoft.Teams.Classic' and installing "Teams 2.0"
https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/Teams 
(as of now the latest version is [24193.1805.3040.8975](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/Teams/24193.1805.3040.8975))
